packages/react/menu/src/Menu.tsx

```diff
@@ -710,8 +743,31 @@ const MenuSubTrigger = React.forwardRef((props, forwardedRef) => {
         data-state={getOpenState(context.open)}
         {...props}
         ref={composeRefs(forwardedRef, context.onTriggerChange)}
-        onMouseMove={composeEventHandlers(props.onMouseMove, () => {
-          if (!props.disabled) context.onOpenChange(true);
+        onMouseMove={composeEventHandlers(props.onMouseMove, (event) => {
+          if (contentContext.isMouseMovingToSubmenu(event)) return;
+          if (!props.disabled && !context.open) {
+            openTimersRef.current.push(window.setTimeout(() => context.onOpenChange(true), 100));
+          }
+        })}
+        onMouseLeave={composeEventHandlers(props.onMouseLeave, (event) => {
+          openTimersRef.current.map(window.clearTimeout);
+          openTimersRef.current = [];
+
+          const contentRect = context.content?.getBoundingClientRect();
+          if (contentRect) {
+            contentContext.onPointerGraceAreaChange([
+              { x: event.clientX, y: event.clientY },
+              { x: contentRect.left, y: contentRect.top },
+              { x: contentRect.left, y: contentRect.bottom },
+            ]);
+
+            pointerGraceDurationTimer.current = window.setTimeout(
+              () => contentContext.onPointerGraceAreaChange(null),
+              300
+            );
+          } else {
+            contentContext.onPointerGraceAreaChange(null);
```

Comment:
I'm curious when we fall into this branch? If the mouse is leaving the submenu trigger then isn't the content always open at that point?

Reply:
There's 100ms where you could leave without opening a submenu.

Reply:
Can we add that as a comment as well please? 🙏